.circleci/config.yml

```diff
@@ -543,6 +543,21 @@ jobs:
       - checkout
       - setup_environment:
           platform: << parameters.platform >>
+      - when:
+          condition:
+            not: << pipeline.parameters.nightly >>
+          steps:
+            - run:
+                command: >
+                  # Validate that version in the Cargo.toml matches the tag version
+                  # Fail the release if the versions don't match
+                  # Note that the tag has the prefix `v`, so this needs to be used to prefix the VERSION variable.
+                  VERSION=v$(cargo metadata --format-version=1 --no-deps | jq --raw-output '.packages[0].version')
```

Comment:
Just as a thought: Is there a reason this can't be an `xtask` command?  I feel like a series of commands that are written (**and tested**) in Rust, including being able to run them locally, is more manageable than just adding more and more to our CircleCI file.

Reply:
Let's talk about this.
I actually don't think xtask is a good build tool in general, it's very high cost to add simple features, adds a ton to our build times and doesn't have many of the basic features that a build tool should have.

I understand the concern about being stuck with circle-ci, but in that case we could consider: https://github.com/sagiegurari/cargo-make